modules/pam_unix/passverify.c

```diff
@@ -103,6 +103,40 @@ verify_pwd_hash(const char *p, char *hash, unsigned int nullok)
 			 * Ok, we don't know the crypt algorithm, but maybe
 			 * libcrypt knows about it? We should try it.
 			 */
+#if defined(CRYPT_CHECKSALT_AVAILABLE) && CRYPT_CHECKSALT_AVAILABLE
+			/*
+			 * Check for hashing methods that are disabled by
+			 * libcrypt configuration and/or system preset.
+			 */
+			if (crypt_checksalt(hash) == CRYPT_SALT_METHOD_DISABLED) {
```

Comment:
Please call the crypt_checksalt just once and assign the result to a variable and then test it. Also do not write the whole hash to the system log - something like hash \"%.3s...\" should be fine.

Reply:
Fixed.